src/track/track.cpp

```diff
@@ -431,17 +433,34 @@ void Track::afterBeatsAndBpmUpdated(
         QMutexLocker* pLock) {
     DEBUG_ASSERT(pLock);
 
-    const auto bpm = getBpmWhileLocked();
     markDirtyAndUnlock(pLock);
-    emitBeatsAndBpmUpdated(bpm);
+    emitBeatsAndBpmUpdated();
 }
 
-void Track::emitBeatsAndBpmUpdated(
-        mixxx::Bpm newBpm) {
-    emit bpmUpdated(newBpm.getValue());
+void Track::emitBeatsAndBpmUpdated() {
+    emit bpmChanged();
     emit beatsUpdated();
 }
 
+void Track::emitMetadataChanged() {
```

Comment:
This reads as if a signal "MetadataChanged" will be emitted. 
How about  emitChangedSignalsForAllMetadata() or something like that.